### ISSUE TITLE
Expose a lua API or ex commands for renaming without nvim-tree?

### ISSUE THREAD

**Issue by IndianBoy42** [antosha417/nvim-lsp-file-operations#13]

**Comment by chrisgrieser** [comment]
Also came here for this. I don't use a file tree plugin at all and installing one just for the LSP renaming operation feels a bit bad

**Comment by antosha417** [comment]
Yeah, I totally agree. Would be a great addition to the plugin. @IndianBoy42 how would you like the api to look like?

I think for now you can just call this function before moving the files around. 
https://github.com/antosha417/nvim-lsp-file-operations/blob/master/lua/lsp-file-operations/will-rename.lua#L23-L38
```lua
require('lsp-file-operations.will-rename').callback({old_name: "path/to/file", new_name: "new/path/to/file"})
```

**Comment by chrisgrieser** [comment]
for the API, something like these seems fine to me?

```lua
require('lsp-file-operations').rename({ old_name = "foo", new_name = "bar"})
```

edit: `old_name` should be optional, defaulting to the current file

**Comment by antosha417** [comment]
@chrisgrieser, do you want it to actually move the files? after calling callback

There should also be a way of specifying which callback you want to call. Right now we have only `will-rename` but lsp spec supports others. Like [did-rename](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_didRenameFiles). Or should we call all of them?

**Comment by antosha417** [comment]
I don't have a lot of time to work on this for now. You can try it in your local config and open a pr if you want. I'll happily merge it

**Comment by bugabinga** [comment]
@chrisgrieser are you planning to integrate this into genghis *wink* *wink* ? :)

**Comment by chrisgrieser** [comment]
@bugabinga No, but I'll gladly accept a PR if someone would *wink wink* :)

**Comment by DrKJeff16** [comment]
@chrisgrieser I'm more than willing to give this a try. Will take a look later on...
